msal/src/Microsoft.Identity.Client/CacheV2/Impl/Jwt.cs

```diff
@@ -0,0 +1,63 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Identity.Client.CacheV2.Impl.Utils;
+using Microsoft.Identity.Core.Helpers;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client.CacheV2.Impl
+{
+    internal class Jwt
+    {
+        public Jwt(string raw)
+        {
+            Raw = raw;
+            if (string.IsNullOrWhiteSpace(raw))
+            {
+                // warning: constructed jwt from empty string
+                return;
+            }
+
+            string[] sections = Raw.Split('.');
+            if (sections.Length != 3)
+            {
+                throw new ArgumentException("failed jwt decode: wrong number of sections", nameof(raw));
```

Comment:
>"failed jwt decode: wrong number of sections" [](start = 44, length = 45)

goal other places has been to keep strings in one file (is easier to review as a whole and it will be easier to make general updates to all errors, messages, etc.

Reply:
All exceptions/logging/errors need a 2nd round cleanup pass under Cachev2/...